File: migration/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

int main(void)
{
    //get text
    string s = get_string("Text:");
    int c = 0;
    int w = 0;
    int p = 0;

    for (int i = 0; strlen(s) > i; i++)
    {
        // letters
        if ((s[i] < 91 && s[i] > 64) || (s[i] < 123 && s[i] > 96))
        {
            c++;
        }
        // words
        if (s[i] == ' ')
        {
            w++;
        }
        // sentences
        if (s[i] == '.' || s[i] == '?' || s[i] == '!')
        {
            p++;
        }

    }
    float x = 0;
    float L = (float)c / (float)(w + 1) * 100;
    float S = (float)p / (float)(w + 1) * 100;
    // index = 0.0588 * L - 0.296 * S - 15.8
    x = 0.0588 * L - 0.296 * S - 15.8;
    int g = round(x);
    if (x > 16)
    {
        printf("Grade 16+\n");
    }
    else if (x < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", g);
    }
}

File: migration/pset6/readability/readability.py
from cs50 import get_string

# main


def main():
    s = get_text()
    let = 0
    wor = 0
    sen = 0
    space = [" "]
    dots = [".", "!", "?"]

# loop

    for c in s:
        if c.isalpha():
            let += 1
        if c in space:
            wor += 1
        if c in dots:
            sen += 1
    x = 0
    L = float(let / float(wor + 1) * 100)
    S = float(sen / float(wor + 1) * 100)

# maths

    x = 0.0588 * L - 0.296 * S - 15.8
    g = str(round(x))

# printing

    if x > 16:
        print("Grade 16+")
    elif x < 1:
        print("Before Grade 1")
    else:
        print("Grade " + g)

# get input


def get_text():
    s = get_string("Text: ")
    return s


# execute
main()